src/App.tsx

```diff
@@ -1,9 +1,23 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState<string>('');
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const visibleMovies = moviesFromServer.filter(movie => {
+    const { title, description } = movie;
+    const lowerCaseQuery = query.toLowerCase();
+
+    return title.toLowerCase().includes(lowerCaseQuery)
+    || description.toLowerCase().includes(lowerCaseQuery);
```

Comment:
maybe this is kinda overkill, you will create this variables on each iteration, maybe I'm wrong and this how it should be